src/concurrency/data_race.rs

```diff
@@ -851,25 +851,29 @@ impl VClockAlloc {
     pub fn read<'tcx>(
         &self,
         alloc_id: AllocId,
-        range: AllocRange,
+        access_range: AllocRange,
         machine: &MiriMachine<'_, '_>,
     ) -> InterpResult<'tcx> {
         let current_span = machine.current_span();
         let global = machine.data_race.as_ref().unwrap();
         if global.race_detecting() {
             let (index, mut clocks) = global.current_thread_state_mut(&machine.threads);
             let mut alloc_ranges = self.alloc_ranges.borrow_mut();
-            for (offset, range) in alloc_ranges.iter_mut(range.start, range.size) {
-                if let Err(DataRace) = range.read_race_detect(&mut clocks, index, current_span) {
+            for (access_subrange, other_range) in
+                alloc_ranges.iter_mut(access_range.start, access_range.size)
```

Comment:
Ah dang, `clocks` already exists here. I think I'll make a PR renaming things in data_race, this is not your problem -- but it will conflict with your PR.

Reply:
Please rebase onto https://github.com/rust-lang/miri/pull/2876.